framework/common/src/main/java/edu/kit/kastel/mcse/ardoco/core/common/util/Comparators.java

```diff
@@ -4,24 +4,94 @@
 import java.util.Collection;
 
 import org.eclipse.collections.api.collection.ImmutableCollection;
+import org.eclipse.collections.api.ordered.SortedIterable;
 
+/**
+ * Provides functions to compare collections regardless of order
+ */
 public final class Comparators {
     private Comparators() {
         throw new IllegalAccessError();
     }
 
+    /**
+     * {@return Whether both collections consist of equal elements (regardless of order)}
+     *
+     * @param first  collection
+     * @param second collection
+     * @param <T>    Type of the collection
+     */
     public static <T> boolean collectionsEqualsAnyOrder(ImmutableCollection<T> first, ImmutableCollection<T> second) {
         return collectionsEqualsAnyOrder(first.castToCollection(), second.castToCollection());
     }
 
+    /**
+     * {@return Whether both collections consist of equal elements (regardless of order)}
+     *
+     * @param first  collection
+     * @param second collection
+     * @param <T>    Type of the collection
+     */
     public static <T> boolean collectionsEqualsAnyOrder(Collection<T> first, Collection<T> second) {
         return first.size() == second.size() && first.containsAll(second) && second.containsAll(first);
     }
 
+    /**
+     * {@return Whether both sorted iterables consist of equal elements (regardless of order)}
+     *
+     * @param first  collection
+     * @param second collection
+     * @param <T>    Type of the collection
+     */
+    public static <T> boolean collectionsEqualsAnyOrder(SortedIterable<T> first, SortedIterable<T> second) {
+        var f = first.iterator();
+        var s = second.iterator();
+        while (f.hasNext() || s.hasNext()) {
+            if (f.hasNext() != s.hasNext())
+                return false;
+            if (!f.next().equals(s.next()))
+                return false;
+        }
+        return true;
+    }
+
+    /**
+     * {@return Whether both collections consist of the same references (regardless of order)}
+     *
+     * @param first  collection
+     * @param second collection
+     * @param <T>    Type of the collection
+     */
     public static <T> boolean collectionsIdentityAnyOrder(ImmutableCollection<T> first, ImmutableCollection<T> second) {
         return collectionsIdentityAnyOrder(first.castToCollection(), second.castToCollection());
     }
 
+    /**
+     * {@return Whether both sorted iterables consist of the same references (regardless of order)}
+     *
+     * @param first  collection
+     * @param second collection
+     * @param <T>    Type of the collection
+     */
+    public static <T> boolean collectionsIdentityAnyOrder(SortedIterable<T> first, SortedIterable<T> second) {
```

Comment:
I am not sure what you mean. This is just an overload of a function that is supposed to check whether two collections contain identical elements. If two sorted iterables contain identical elements, the statement "Both collections contain identical elements in any order" still holds. This overload is merely for the special case where the provided collections are already sorted, because it speeds up the check. Or am I missing something? I don't see a scenario where this is a problem

I did however spot a bug just now. Are the functions supposed to ignore cardinality? Because as of now, the default implementation will return true for two collections such as:

Coll A: A-A-B-C
Coll B: A-B-B-C

Im not sure that this is intended by the original author of the functions, but if it is I need to adapt the overload slightly